server/gfmd/user.c

```diff
@@ -1966,44 +2001,21 @@ gfm_server_user_auth_modify(struct peer *peer,
 		gflog_debug(GFARM_MSG_UNFIXED, "%s (%s@%s) %s: %s",
 			diag, peer_get_username(peer), peer_get_hostname(peer),
 			username, gfarm_error_string(e));
-	} else if ((e = user_info_verify(&u->ui, diag)) != GFARM_ERR_NO_ERROR) {
-		gflog_debug(GFARM_MSG_1003462,
-			"%s: user_info_verify: %s",
-			diag, gfarm_error_string(e));
 	} else if (gfarm_read_only_mode()) {
-		gflog_debug(GFARM_MSG_1005143, "%s (%s@%s) during read_only",
+		gflog_debug(GFARM_MSG_UNFIXED, "%s (%s@%s) during read_only",
 			diag, peer_get_username(peer), peer_get_hostname(peer));
 		e = GFARM_ERR_READ_ONLY_FILE_SYSTEM;
-	} else if ((e = rpc_name_with_tenant(peer, from_client,
-			&name_with_tenant, &process, diag))
-			!= GFARM_ERR_NO_ERROR) {
-		/* nothing to do */
-	} else if ((u = user_tenant_lookup(username)) == NULL) {
-		e = GFARM_ERR_NO_SUCH_USER;
 	} else
 		e = GFARM_ERR_NO_ERROR;
 
 	if (e == GFARM_ERR_NO_ERROR) {
 
 		if (auth_user_id == NULL || strcmp(auth_user_id, "") == 0) {
-			struct db_user_auth_remove_arg arg = {
-				username,
-				auth_user_id_type_str
-			};
-			user_auth_id_remove(u, auth_user_id_type_str);
-
-			struct user_auth_key key = {
-				auth_user_id_type,
-				auth_user_id
-			};
-			gfarm_hash_purge(user_id_map, &key, sizeof(key));
+			e = user_auth_id_remove(u, auth_user_id_type_str);
 		} else {
-			user_auth_id_modify(u,
+			e = user_auth_id_modify(u,
 				auth_user_id_type_str, auth_user_id);
-			user_enter_auth_id(auth_user_id_type, auth_user_id, u);
 		}
-		free(u->auth_user_id[auth_user_id_type]);
-		u->auth_user_id[auth_user_id_type] = strdup(auth_user_id);
 	} else
 		free(auth_user_id);
```

Comment:
there is a memory leak here.
remove "else free(auth_user_id);", and add "free(auth_user_id);" after "free(auth_user_id_type_str);"